h/static/scripts/controllers.coffee

```diff
@@ -48,7 +48,8 @@ class AppController
           annotationMapper.loadAnnotations data
         when 'delete'
           for annotation in data
-            $scope.$emit('annotationDeleted', annotation)
+            if a = threading.idTable[annotation.id]?.message
+              $scope.$emit('annotationDeleted', a)
```

Comment:
This is the right fix, and consistent with the how the controllers work at the moment.

`annotationMapper.deleteAnnotation` doesn't work either as that is intended to delete an annotation, this one has already been deleted.

Really though the streamer or a wrapper around the streamer should be looking up local annotations in the cache before they get to the controller. Threading should have a public `get()` method or the cache should be moved elsewhere as @tilgovi has already suggested. Lastly these events should be moved into a service so we don't have a million places with hard coded event strings.

This is all a bit much for a single PR, but I'd rather we didn't just add this here as is.